freelist.go

```diff
@@ -171,7 +174,11 @@ func (f *freelist) read(p *page) {
 	copy(f.ids, ids)
 
 	// Make sure they're sorted.
-	sort.Sort(pgids(f.ids))
+	for i, v := range f.ids[:len(f.ids)-1] {
+		if f.ids[i+1] < v {
+			sort.Sort(pgids(f.ids))
+		}
```

Comment:
I think you want to `break` here, else you'll sort, then keep testing all following indices.